src/DurableTask.Netherite/OrchestrationService/NetheriteOrchestrationServiceSettings.cs

```diff
@@ -28,22 +29,11 @@ public class NetheriteOrchestrationServiceSettings
 
         /// <summary>
         /// Gets or sets the name for resolving the Eventhubs namespace connection string.
-        /// Alternatively, may contain special strings specifying the use of the emulator.
+        /// Pseudo-connection-strings "Memory" or "SingleHost" can be used to configure
+        /// in-memory emulation, or single-host configuration, respectively.
         /// </summary>
         public string EventHubsConnectionName { get; set; } = "EventHubsConnection";
 
-        /// <summary>
-        /// The resolved storage connection string. Is never serialized or deserialized.
-        /// </summary>
-        [JsonIgnore]
-        public string ResolvedStorageConnectionString { get; set; }
```

Comment:
There are several breaking changes in this file. Are we okay with that? If we follow semver, that would require a major version rev.

The alternative is to try and do additive changes only - retaining old functionality as is and only adding it what you need for the new functionality.

Reply:
I can see how much work is required to maintain compatibility. Perhaps keeping some deprecated ResolvedConnectionString fields is enough.

Note that this affects only DTFx users, for DF users there are no breaking changes.

Reply:
I will add a compatibility shim so there are no breaking changes.

Reply:
Yeah, avoiding breaking changes is very tricky. Here is a list of what is and is not allowed: https://learn.microsoft.com/en-us/dotnet/core/compatibility/

Reply:
To avoid breaking changes in this PR, I suggest making sure everything is purely additive. Do not remove or change existing method signatures or properties - only add new ones. The implementations then need to make decisions at runtime for which property to use. You _can_ update the getting/setting implementations though. So, an example of what you are trying to update could be like:

Before:
``` CSharp
public string OldConnectionString { get; set; }
``` 

After:
``` CSharp
public ComplexConnectionType { get; set; }

public string OldConnectionString
{
    get => ComplexConnectionType?.ConnectionString;
    set => ComplexConnectionType = new ComplexConnectiontype(value); // also need some logic around going from non-null to null
}

public class ComplexConnectionType
{
    public ComplexConnectionType(string connectionString) { } // a ctor for the connection string scenario

    public ComplexConnectionType(TokenCredential credential) { } // a ctor for the new recommended way. Just an example implementation - format this however you need.
```

Reply:
We can then remove the shim when we go to 2.0

Reply:
There are no more breaking changes here.